extensions-core/datasketches/src/main/java/io/druid/query/aggregation/datasketches/tuple/ArrayOfDoublesSketchMultiPostAggregator.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.query.aggregation.datasketches.tuple;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import io.druid.query.aggregation.PostAggregator;
+
+/**
+ * Base class for post aggs taking multiple sketches as input
+ */
+public abstract class ArrayOfDoublesSketchMultiPostAggregator extends ArrayOfDoublesSketchPostAggregator
+{
+
+  private final List<PostAggregator> fields;
+
+  @JsonCreator
+  public ArrayOfDoublesSketchMultiPostAggregator(final String name, final List<PostAggregator> fields)
+  {
+    super(name);
+    this.fields = fields;
+  }
+
+  @Override
+  public Set<String> getDependentFields()
+  {
+    final Set<String> dependentFields = super.getDependentFields();
+    for (final PostAggregator field : fields) {
+      dependentFields.addAll(field.getDependentFields());
+    }
+    return dependentFields;
+  }
+
+  @JsonProperty
+  public List<PostAggregator> getFields()
+  {
+    return fields;
+  }
+
+  @Override
+  public String toString()
+  {
+    return this.getClass().getSimpleName() + "{"
+        + "name='" + getName() + '\''
+        + ", fields=" + fields
+        + "}";
+  }
+
+  @Override
+  public boolean equals(final Object o)
+  {
+    if (!super.equals(o)) {
+      return false;
+    }
+    final ArrayOfDoublesSketchMultiPostAggregator that = (ArrayOfDoublesSketchMultiPostAggregator) o;
```

Comment:
It breaks symmetry of equals() and could be subject to ClassCastException.

Reply:
No. That super.equals includes if (getClass() != o.getClass()) return false

Reply:
Then it breaks the substitution principle.

Independent from that, relying on `super.equals()` to check specifically `getClass()` rather than `instanceof` is fragile. Somebody could refactor the superclass and it will break. Add class check in the beginning of this method anyway.

Reply:
@AlexanderSaydakov I was wrong when I've written "Then it breaks the substitution principle", because the superclass is abstract. Actually we should preserve `getClass()` in _abstract_ classes in this hierarchy, so that instances of different concrete subclasses that don't override `equals()` don't appear "equal" to each other.

Could you please additionally add equals() and hashCode() unit tests to all concrete PostAggregator classes, added in this PR, testing -
- equals case (and correspondence with hash code)
- not equals, same class case
- not equals, different classes both not overriding equals() of their abstract superclass
- equals to null (should return false)